matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/permalinks/PermalinkFactory.kt

```diff
@@ -105,6 +105,28 @@ internal class PermalinkFactory @Inject constructor(
                 ?.substringBeforeLast("?")
     }
 
+    fun createHtmlMentionSpanTemplate(forceMatrixTo: Boolean): String {
```

Comment:
I would combine `createMdMentionSpanTemplate` and `createHtmlMentionSpanTemplate` and pass `MENTION_SPAN_TO_HTML_TEMPLATE_BEGIN` or  `MENTION_SPAN_TO_MD_TEMPLATE_BEGIN`  as parameter to avoid code duplication.

Reply:
Fair point. I actually made an enum at the interface level so it's not bind to what's in the factory :)

Reply:
Nice!